src/lib/draw/route/RouteMode.svelte

```diff
@@ -61,11 +68,58 @@
       changeMode
     );
   });
+
+  async function fetchWithProgress(url, setProgress: (number) => void) {
```

Comment:
```suggestion
  // setProgress will generally be called with a number between 0 and 100, but it may exceed this for gzipped files.
  // The server must send back a Content-Length header.
  async function fetchWithProgress(url: string, setProgress: (number) => void) {
```